### ISSUE TITLE
ExpressionsValidator

### ISSUE THREAD

**Issue by lgcardeas** [LorenzoBettini/packtpub-xtext-book-examples#3]
How can i do to initialize a variable with 0 if it wasn't declared before?

**Comment by LorenzoBettini** [comment]
I don't understand your question: please give me more context.

**Comment by lgcardeas** [comment]
I want to change your example "expressions". I want a variable has a value zero (0) by default when it is used in right part of the expression and it is not used in the left part yet.

**Comment by LorenzoBettini** [comment]
Mh... so you want a DSL where variables have not to be explicitly declared, right?

That's not trivial to implement, since scoping and references assume that you have an explicit declaration to refer to; not to mention that what you try to implement is oriented to a dynamically typed language, not a statically typed one. You should change many things, starting from the grammar itself since on the right hand side of an expression you don't have {VariableRef} variable=[Variable] but you could have also a declaration; even worse, you can't know in advance (i.e., in the grammar) whether an ID is a declaration or a reference to a declaration: you can tell that only when the program is parsed. Scoping has to be heavily adapted as well...

In general, I always implement statically typed DSL (and Xtext aims at statically typed DSL, though, of course, can be used also for Dynamically typed DSLs, but with additional customizations).

Long story short: do you really need to implement such a DSL? Or are you just experimenting?

**Comment by LorenzoBettini** [comment]
I'm closing this issue since I haven't received any further feedback.
